From 92046843a626ecd3a87ce86fac9551481cbc7d45 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 28 Mar 2023 12:06:30 +0200
Subject: [PATCH 294/954] feat(docs): add github links

NOT_UPSTREAMABLE

The ST wiki links to docs files by default to github pages.
Add github links in the updated pages where :ref: is used to link to
readthedocs pages.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git d7c16ac9a4b7bf44f5e2705632002058a2a2c962 feat(docs): add github links]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I9a82f830fe4b4d11531da2900d3f2a4910fc9696
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/297703
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 docs/plat/st/stm32mp1.rst  | 3 ++-
 docs/plat/st/stm32mp2.rst  | 4 +++-
 docs/plat/st/stm32mpus.rst | 5 ++++-
 docs/plat/stm32mp1.rst     | 7 +++++--
 4 files changed, 14 insertions(+), 5 deletions(-)

diff --git a/docs/plat/st/stm32mp1.rst b/docs/plat/st/stm32mp1.rst
index b6e4b0d84a..40b4987486 100644
--- a/docs/plat/st/stm32mp1.rst
+++ b/docs/plat/st/stm32mp1.rst
@@ -7,7 +7,7 @@ It is an Armv7-A platform, using dedicated code from TF-A.
 More information can be found on `STM32MP1 Series`_ page.
 
 For TF-A common configuration of STM32 MPUs, please check
-:ref:`STM32 MPUs` page.
+:ref:`STM32 MPUs` page or `Github STM32 MPUs`_.
 
 STM32MP1 Versions
 -----------------
@@ -216,5 +216,6 @@ __________________
 
 .. _STM32MP1 Series: https://www.st.com/en/microcontrollers-microprocessors/stm32mp1-series.html
 .. _STM32MP1 part number codification: https://wiki.st.com/stm32mpu/wiki/STM32MP15_microprocessor#Part_number_codification
+.. _Github STM32 MPUs: https://github.com/STMicroelectronics/arm-trusted-firmware/tree/HEAD/docs/plat/st/stm32mpus.rst
 
 *Copyright (c) 2023, STMicroelectronics - All Rights Reserved*
diff --git a/docs/plat/st/stm32mp2.rst b/docs/plat/st/stm32mp2.rst
index 43e131d714..f10de1d09a 100644
--- a/docs/plat/st/stm32mp2.rst
+++ b/docs/plat/st/stm32mp2.rst
@@ -5,7 +5,7 @@ STM32MP2 is a microprocessor designed by STMicroelectronics
 based on Arm Cortex-A35.
 
 For TF-A common configuration of STM32 MPUs, please check
-:ref:`STM32 MPUs` page.
+:ref:`STM32 MPUs` page or `Github STM32 MPUs`_.
 
 STM32MP2 Versions
 -----------------
@@ -130,4 +130,6 @@ ___
         BL32_EXTRA1=<optee_directory>/tee-pager_v2.bin
         fip
 
+.. _Github STM32 MPUs: https://github.com/STMicroelectronics/arm-trusted-firmware/tree/HEAD/docs/plat/st/stm32mpus.rst
+
 *Copyright (c) 2023, STMicroelectronics - All Rights Reserved*
diff --git a/docs/plat/st/stm32mpus.rst b/docs/plat/st/stm32mpus.rst
index 931dd57fe1..63840bdb3f 100644
--- a/docs/plat/st/stm32mpus.rst
+++ b/docs/plat/st/stm32mpus.rst
@@ -4,7 +4,7 @@ STM32 MPUs
 STM32 MPUs are microprocessors designed by STMicroelectronics
 based on Arm Cortex-A. This page presents the common configuration of STM32
 MPUs, more details and dedicated configuration can be found in each STM32 MPU
-page (:ref:`STM32MP1` or :ref:`STM32MP2`)
+page (:ref:`STM32MP1` (`Github STM32MP1`_) or :ref:`STM32MP2` (`Github STM32MP2`_) )
 
 Design
 ------
@@ -73,6 +73,9 @@ It should contain at least those partitions:
 
 Usually, two copies of fsbl are used (fsbl1 and fsbl2) instead of one partition fsbl.
 
+.. _Github STM32MP1: https://github.com/STMicroelectronics/arm-trusted-firmware/tree/HEAD/docs/plat/st/stm32mp1.rst
+.. _Github STM32MP2: https://github.com/STMicroelectronics/arm-trusted-firmware/tree/HEAD/docs/plat/st/stm32mp2.rst
+
 --------------
 
 *Copyright (c) 2023, STMicroelectronics - All Rights Reserved*
diff --git a/docs/plat/stm32mp1.rst b/docs/plat/stm32mp1.rst
index f2c8fd2281..2b99f5e681 100644
--- a/docs/plat/stm32mp1.rst
+++ b/docs/plat/stm32mp1.rst
@@ -3,8 +3,11 @@
 STMicroelectronics STM32MP1 (old page)
 ======================================
 
-Please check :ref:`STM32 MPUs` page for generic information about
+Please check :ref:`STM32 MPUs` page  or `Github STM32 MPUs`_ for generic information about
 STMicroelectronics STM32 microprocessors in TF-A, and :ref:`STM32MP1` page
-for specificities on STM32MP1x platforms.
+or `Github STM32MP1`_ for specificities on STM32MP1x platforms.
+
+.. _Github STM32 MPUs: https://github.com/STMicroelectronics/arm-trusted-firmware/tree/HEAD/docs/plat/st/stm32mpus.rst
+.. _Github STM32MP1: https://github.com/STMicroelectronics/arm-trusted-firmware/tree/HEAD/docs/plat/st/stm32mp1.rst
 
 *Copyright (c) 2018-2023, STMicroelectronics - All Rights Reserved*
-- 
2.53.0

